### DISCUSSION TITLE
Train YOLOv8 with SAHI on custom dataset

### DISCUSSION THREAD

**Discussion by Mu-Magdy** [obss/sahi#1059] [Q&A] [answered]
Hi There,

I can't fully comprehend how to train my custom data with yolov8 weights and sahi, is it feasible ?
My data is on roboflow and i want to use yolov8x
I trained my data using yolov8x but it didn't get high scores as my objects are really small how to integrate sahi with it

**Comment by fcakyon** [comment, marked as answer]
Hello @Mu-Magdy,

You can slice a COCO formatted dataset as:

```python
from sahi.slicing import slice_coco

coco_dict, coco_path = slice_coco(
    coco_annotation_file_path=coco_annotation_file_path,
    image_dir=image_dir,
    slice_height=256,
    slice_width=256,
    overlap_height_ratio=0.2,
    overlap_width_ratio=0.2,
)
```

then use this dataset during training.

**Comment by scchess** [comment]
Once trained, you'd then use the SAHI's predictions.
